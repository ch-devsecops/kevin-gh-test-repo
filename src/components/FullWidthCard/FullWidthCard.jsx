import React from 'react';
import PropTypes from 'prop-types';
import {
  Box,
  MarkdownHeading,
  Markdown,
  MediaItem,
  useOptionalVideo,
  Optional,
} from '@honda-canada/design-system-react';
import themeStyles from './FullWidthCard.styles';

const CTAGroup = themeStyles.apply(Box, 'CTAGroup');
const ContentContainer = themeStyles.apply(Box, 'ContentContainer');
const ContentWrapper = themeStyles.apply(Box, 'ContentWrapper');

const FullWidthCardBase = ({
  title,
  bodyText,
  image,
  video,
  isTertiaryCTAGroup,
  ctas,
  anchorId,
  gtmTags = {},
  isDark,
}) => {
  const { optionalVideo } = useOptionalVideo(video);

  return (
    <Box
      pb={['l', 'big']}
      id={anchorId}
      data-gtm-component-type={gtmTags.type}
      data-gtm-category={gtmTags.category}
      backgroundColor={isDark ? 'black' : 'white'}
    >
      <MediaItem image={image} video={optionalVideo} containerSize="100%" />

      <Optional when={title || bodyText}>
        <ContentContainer>
          <ContentWrapper>
            <Optional when={title}>
              <MarkdownHeading headingOverride="h5" mb={['xs', 'm']} color={isDark ? 'white' : undefined}>
                {title}
              </MarkdownHeading>
            </Optional>

            <Optional when={bodyText}>
              <Markdown textAlign="center" mb={['0']} color={isDark ? 'white' : undefined}>
                {bodyText}
              </Markdown>
            </Optional>
          </ContentWrapper>

          <CTAGroup isTertiary={isTertiaryCTAGroup}>{ctas}</CTAGroup>
        </ContentContainer>
      </Optional>
    </Box>
  );
};

FullWidthCardBase.defaultProps = {
  bodyText: '',
  isTertiaryCTAGroup: false,
};

FullWidthCardBase.propTypes = {
  title: PropTypes.string,
  /**
   * Markdown enabled text
   */
  bodyText: PropTypes.string,
  isTertiaryCTAGroup: PropTypes.bool,
  image: PropTypes.element,
  video: PropTypes.element,
  ctas: PropTypes.arrayOf(PropTypes.element),
  anchorId: PropTypes.string,
  gtmTags: PropTypes.shape({
    category: PropTypes.string,
    type: PropTypes.string,
  }),
  isDark: PropTypes.bool,
};

export default FullWidthCardBase;
