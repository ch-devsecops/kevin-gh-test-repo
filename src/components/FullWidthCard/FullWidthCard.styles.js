import css from '@styled-system/css';
import { ThemeStyles } from '@honda-canada/design-system-react';

const themeStyles = new ThemeStyles();

themeStyles.addBaseStyles('ContentContainer')(
  css({
    mt: ['default', 'l'],
    px: ['20px', 'm'],
    textAlign: 'center',
  }),
);

themeStyles.addBaseStyles('ContentWrapper')(
  css({
    maxWidth: '824px',
    margin: '0 auto',
  }),
);

themeStyles.addBaseStyles('CTAGroup')(({ isTertiary }) => {
  const mobileDirection = isTertiary ? 'row' : 'column';
  const desktopTopMargin = isTertiary ? 'm' : 'l';
  const mobileTopMargin = isTertiary ? 'm' : 'default';

  const styles = {
    textAlign: 'center',
    display: 'inline-flex',
    flexDirection: [mobileDirection, 'row'],
    mt: [mobileTopMargin, desktopTopMargin],

    '& > *': {
      mx: 'xs',
    },
  };

  if (!isTertiary) {
    styles['& > *'] = {
      mx: 'xs',
      mb: ['m', 0],
    };

    styles['& > *:first-child, & > *:last-child'] = {
      mb: ['m', 0],
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
    };
  }

  return css(styles);
});

export default themeStyles;
