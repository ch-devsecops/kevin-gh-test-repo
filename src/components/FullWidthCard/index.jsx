import React, { useContext } from 'react';
import { Image } from '@honda-canada/design-system-react';
import FullWidthCardBase from './FullWidthCard';
import { getVideoProps, mapGTMCategory, mapJssFieldsToCtaComponents } from '../../utils/sitecoreFields';
import { ModelExplorationContext } from '../ModelExplorationContext';

const FullWidthCard = ({ fields, rendering }) => {
  const modelExplorationContext = useContext(ModelExplorationContext) || {};
  const { isDark } = modelExplorationContext;
  if (!fields) return null;
  const { anchorId, title, bodyText, mediaImage, gtmTitle, gtmCategory, videoUrl } = fields;
  const ctas = mapJssFieldsToCtaComponents(fields, gtmTitle?.value);
  const gtmTags = {
    category: mapGTMCategory(gtmCategory),
    title: gtmTitle?.value,
    type: rendering?.componentName,
  };

  return (
    <FullWidthCardBase
      title={title?.value}
      bodyText={bodyText?.value}
      ctas={title?.value && bodyText?.value ? ctas : null}
      anchorId={anchorId?.value}
      gtmTags={gtmTags}
      image={<Image {...mediaImage?.value} />}
      video={getVideoProps(videoUrl?.value)}
      isDark={isDark}
    />
  );
};

export default FullWidthCard;
