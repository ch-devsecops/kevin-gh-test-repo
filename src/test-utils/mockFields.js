/* eslint-disable max-len */
/* eslint-disable max-classes-per-file */

// FIXME: jsdoc popup not working on test files
/** Class representing a mock data. */
class MockData {
  /**
   * Add a field.
   * @param {string} key - key of the field obj: i.e. 'title'
   * @param {Object} value - value of the field obj: i.e. { value: 'some text' }.
   */
  add(key, value) {
    this[key] = value;
  }
}

/**
 * Class representing mock fields object.
 * @extends MockData
 *
 * available method: add()
 *
 * default fields: [anchorId, title, bodyText, image, ctaLink, ctaType, ctaIcon, gtmCategory, gtmTitle, contentTitle, subtitle]
 *
 * note: we can add more commonly used fields in the class declaration if needed
 */
class MockFields extends MockData {
  constructor() {
    super();

    this.title = { value: 'Title' };
    this.bodyText = { value: 'BodyText' };
    this.image = {
      value: { src: 'mock.png', alt: 'mock alt' },
    };
    this.ctaLink = {
      value: {
        href: 'www.source.ca',
        title: 'cta aria label',
        text: 'CTA Text',
      },
    };
    this.ctaType = {
      value: 'Tertiary',
    };
    this.ctaIcon = {
      fields: {
        value: {
          value: 'arrowRight',
        },
      },
    };
    this.gtmCategory = {
      fields: {
        value: {
          value: 'Others',
        },
      },
    };
    this.gtmTitle = {
      value: 'gtm title',
    };
    this.contentTitle = {
      value: 'Content Title',
    };
    this.subtitle = {
      value: 'Content Subtitle',
    };
    this.seoH1 = {
      value: '',
    };
    this.anchorId = {
      value: 'anchor-id',
    };
    this.ctaLabel = {
      value: 'CTA Label',
    };
  }
}

/**
 * Class representing a mock rendering object
 * @extends MockData
 *
 * available method: add()
 */
class MockRendering extends MockData {
  /**
   * Create a mock rendering object
   * @param {string} componentName - adds componentName
   */
  constructor(componentName) {
    super();

    this.componentName = componentName;
  }
}

export { MockFields, MockRendering };
