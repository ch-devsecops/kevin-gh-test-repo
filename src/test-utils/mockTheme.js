/**
 * Mock styled-components theme, if you want to test how your component
 * implements theme styles. Import as-is, or spread in any
 * additional values you need.
 *
 * Example usage:

  it('should apply the bgColor param', () => {
    const { colors } = MockTheme;
    const { container } = render(
      <MyComponent
        fields={mockFields}
        params={{ bgColor: 'Gray' }}
      />,
    );
    const wrapper = container.querySelector('section');

    expect(wrapper).toHaveStyleRule('background-color', colors.grey[5]);
  });
 */

import { HONDA_THEME_NAME } from '../utils/constants';

const mockTheme = {
  letterSpacings: {
    sm: '0.5px',
    md: '1px',
  },
  lineHeights: {
    sm: '22px',
    md: '26px',
  },
  colors: {
    white: '#fff',
    transparent: 'transparent',
    grey: ['#707070', '#C7C7C7', '#D5D5D5', '#EAEAEA', '#F8F8F8', '#F5F5F5'],
    black: '#2C2C2C',
    red: '#DA2C2E',
    redShades: ['rgba(200, 16, 46, 0.1)'],
    aodaRed: '#D92629',
    lightRed: '#EC9597',
    darkRed: '#A21C1D',
    blue: '#334C96',
    darkBlue: '#002D62',
    lightBlue: '#00447826',
    yellow: ['#F9CA52', '#EDAB00', '#8D763F'],
    fadedBlue: '#ADB7D5',
    primary: '#DA2C2E',
    primaryAoda: '#D92629',
    primaryDark: '#A21C1D',
    primaryLight: '#EC9597',
    secondary: '#2C2C2C',
    secondaryLight: '#C7C7C7',
    tertiary: '#002D62',
    tertiaryLight: '#334C96',
    tertiaryDisabled: '#ADB7D5',
    danger: '#DA2C2E',
    warning: '#2C2C2C',
    warningLight: '#EAEAEA',
    success: '#334C96',
    successLight: '#d9e0e7',
    disabled: '#C7C7C7',
    typographyDefault: '#2C2C2C',
  },
  space: [0, 4, 8, 12, 16, 24, 32, 48, 64, 96, 128, 144, 192, 256],
  borders: [0, '1px solid', '2px solid', '3px solid'],
  breakpoints: ['768px', '992px', '1200px'],
  fontWeights: {
    regular: 'normal',
    bold: 700,
  },
  fontSizes: {
    sm: '12px',
  },
  shadows: {
    smooth: {
      md: '0.1px 0.3px 0.3px rgba(0, 0, 0, 0.02),0.3px 0.7px 0.9px rgba(0, 0, 0, 0.028),0.7px 1.4px 1.8px rgba(0, 0, 0, 0.035),1.5px 2.9px 3.7px rgba(0, 0, 0, 0.047),4px 8px 10px rgba(0, 0, 0, 0.07)',
    },
    box: {
      subNav: '0px 2px 4px rgba(0, 0, 0, 0.1)',
    },
  },
  zIndices: {
    accordionItemTitle: 200,
    modal: 13001,
    sectionNav: 13000,
  },
  transitionDuration: {
    t1: '1s',
  },
  transitionTimingFunction: {
    default: 'ease-in-out',
    out: 'ease-out',
    in: 'ease-in',
  },
  name: HONDA_THEME_NAME,
  fonts: {
    default: 'Helvetica',
    bold: 'Cooper Black',
    heading: 'Roboto',
  },
  header: {
    desktop: {},
    mobile: {},
  },
  icons: {},
  logIconWarnings: false,
};

export default mockTheme;
