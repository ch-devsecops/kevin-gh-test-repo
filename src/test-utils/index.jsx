import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
