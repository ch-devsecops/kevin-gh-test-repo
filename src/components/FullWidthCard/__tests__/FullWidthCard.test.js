import React from 'react';
import 'jest-styled-components';
import { jssRender, cleanup, MockRendering, MockFields, axe } from 'test-utils';
import FullWidthCard from '../index';

afterEach(cleanup);

const mockFields = new MockFields();
mockFields.add('mediaImage', { value: { src: 'mock.png', alt: 'mock alt' } });
const mockRendering = new MockRendering('FullWidthCard');

describe('<FullWidthCard>', () => {
  const renderMock = () => jssRender(<FullWidthCard fields={mockFields} rendering={mockRendering} />);

  it('should have no accessibility violations', async () => {
    const { container } = renderMock();
    expect(await axe(container)).toHaveNoViolations();
  });

  it('should match the snapshot', async () => {
    const { container } = renderMock();
    expect(container).toMatchSnapshot();
  });
});

/* clarification concerning v3.1.0-develop.145:

  at the time of writing this test, the video URL was not utilized
  in any of the FullWidthCard items, thus it was unnecessary to
  test of having it in the mock fields. Should this change in future
  use `add(key,value)` method in MockFields test utility class.
*/
