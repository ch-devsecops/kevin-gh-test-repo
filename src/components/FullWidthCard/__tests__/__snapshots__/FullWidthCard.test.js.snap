// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullWidthCard> should match the snapshot 1`] = `
.c0 {
  padding-bottom: l;
  background-color: #fff;
}

.c1 {
  width: 100%;
  height: 100%;
  position: relative;
  height: inherit;
  width: inherit;
  width: 100%;
  height: 100%;
}

.c2 {
  display: block;
  width: 100%;
  height: 100%;
  position: relative;
  object-fit: cover;
  object-position: center;
  -webkit-user-drag: none;
  -moz-user-drag: none;
  -ms-user-drag: none;
  user-drag: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3 {
  margin-top: default;
  padding-left: 20px;
  padding-right: 20px;
  text-align: center;
}

.c4 {
  max-width: 824px;
  margin: 0 auto;
}

.c5 {
  font-size: 14px;
  line-height: 22px;
  font-family: Helvetica;
  color: #2C2C2C;
  margin-bottom: xs;
}

.c5 > :last-child {
  margin-bottom: 0;
}

.c5 .override {
  font-size: 14px;
  line-height: 22px;
  font-family: Helvetica;
  text-transform: none;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.c5 p {
  margin-bottom: default;
}

.c5 blockquote {
  margin-bottom: default;
  font-family: Roboto;
  font-size: 24px;
  line-height: 28px;
  quotes: '“' '”';
}

.c5 blockquote p:after {
  content: close-quote;
}

.c5 blockquote p:before {
  content: open-quote;
}

.c5 small {
  font-size: 10px;
  line-height: 14px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.c5 strong {
  font-family: Cooper Black;
  font-weight: normal;
}

.c5 em {
  font-style: italic;
}

.c5 ul {
  list-style-type: disc !important;
  list-style: initial;
}

.c5 ol {
  list-style-type: decimal;
}

.c5 h1 {
  color: inherit;
  font-family: Roboto;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-size: 36px;
  line-height: 40px;
}

.c5 h2 {
  color: inherit;
  font-family: Roboto;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-size: 32px;
  line-height: 36px;
}

.c5 h3 {
  color: inherit;
  font-family: Roboto;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-size: 28px;
  line-height: 32px;
}

.c5 h4 {
  color: inherit;
  font-family: Roboto;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-size: 24px;
  line-height: 28px;
}

.c5 h5 {
  color: inherit;
  font-family: Cooper Black;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-size: 20px;
  line-height: 24px;
}

.c5 h6 {
  color: inherit;
  font-family: Cooper Black;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-size: 16px;
  line-height: 20px;
}

.c5 ul,
.c5 ol {
  list-style-position: inside;
  margin-bottom: default;
}

.c5 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #DA2C2E;
  font-weight: 700;
}

.c5 a:focus {
  border: transparent;
}

.c6 {
  color: #2C2C2C;
  font-family: Roboto;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-size: 20px;
  line-height: 24px;
  font-family: Cooper Black;
}

.c7 {
  font-size: 14px;
  line-height: 22px;
  font-family: Helvetica;
  color: #2C2C2C;
  text-align: center;
  margin-bottom: 0;
}

.c7 > :last-child {
  margin-bottom: 0;
}

.c7 .override {
  font-size: 14px;
  line-height: 22px;
  font-family: Helvetica;
  text-transform: none;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.c7 p {
  margin-bottom: default;
}

.c7 blockquote {
  margin-bottom: default;
  font-family: Roboto;
  font-size: 24px;
  line-height: 28px;
  quotes: '“' '”';
}

.c7 blockquote p:after {
  content: close-quote;
}

.c7 blockquote p:before {
  content: open-quote;
}

.c7 small {
  font-size: 10px;
  line-height: 14px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.c7 strong {
  font-family: Cooper Black;
  font-weight: normal;
}

.c7 em {
  font-style: italic;
}

.c7 ul {
  list-style-type: disc !important;
  list-style: initial;
}

.c7 ol {
  list-style-type: decimal;
}

.c7 h1 {
  color: inherit;
  font-family: Roboto;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-size: 36px;
  line-height: 40px;
}

.c7 h2 {
  color: inherit;
  font-family: Roboto;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-size: 32px;
  line-height: 36px;
}

.c7 h3 {
  color: inherit;
  font-family: Roboto;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-size: 28px;
  line-height: 32px;
}

.c7 h4 {
  color: inherit;
  font-family: Roboto;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-size: 24px;
  line-height: 28px;
}

.c7 h5 {
  color: inherit;
  font-family: Cooper Black;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-size: 20px;
  line-height: 24px;
}

.c7 h6 {
  color: inherit;
  font-family: Cooper Black;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-size: 16px;
  line-height: 20px;
}

.c7 ul,
.c7 ol {
  list-style-position: inside;
  margin-bottom: default;
}

.c7 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #DA2C2E;
  font-weight: 700;
}

.c7 a:focus {
  border: transparent;
}

.c8 {
  text-align: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: default;
}

.c8 > * {
  margin-left: xs;
  margin-right: xs;
  margin-bottom: m;
}

.c8 > *:first-child,
.c8 > *:last-child {
  margin-bottom: m;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media screen and (min-width:768px) {
  .c0 {
    padding-bottom: big;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    margin-top: l;
    padding-left: m;
    padding-right: m;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    font-size: 16px;
    line-height: 26px;
  }
}

@media screen and (min-width:768px) {
  .c5 .override {
    font-size: 16px;
    line-height: 26px;
  }
}

@media screen and (min-width:768px) {
  .c5 p {
    margin-bottom: l;
  }
}

@media screen and (min-width:768px) {
  .c5 blockquote {
    margin-bottom: l;
    font-size: 40px;
    line-height: 50px;
  }
}

@media screen and (min-width:768px) {
  .c5 h1 {
    font-size: 68px;
    line-height: 74px;
  }
}

@media screen and (min-width:768px) {
  .c5 h2 {
    font-size: 56px;
    line-height: 64px;
  }
}

@media screen and (min-width:768px) {
  .c5 h3 {
    font-size: 44px;
    line-height: 50px;
  }
}

@media screen and (min-width:768px) {
  .c5 h4 {
    font-size: 32px;
    line-height: 38px;
  }
}

@media screen and (min-width:768px) {
  .c5 h5 {
    font-size: 24px;
    line-height: 28px;
  }
}

@media screen and (min-width:768px) {
  .c5 h6 {
    font-size: 18px;
    line-height: 22px;
  }
}

@media screen and (min-width:768px) {
  .c5 ul,
  .c5 ol {
    margin-bottom: l;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    margin-bottom: m;
  }
}

@media screen and (min-width:768px) {
  .c6 {
    font-size: 24px;
    line-height: 28px;
  }
}

@media screen and (min-width:768px) {
  .c7 {
    font-size: 16px;
    line-height: 26px;
  }
}

@media screen and (min-width:768px) {
  .c7 .override {
    font-size: 16px;
    line-height: 26px;
  }
}

@media screen and (min-width:768px) {
  .c7 p {
    margin-bottom: l;
  }
}

@media screen and (min-width:768px) {
  .c7 blockquote {
    margin-bottom: l;
    font-size: 40px;
    line-height: 50px;
  }
}

@media screen and (min-width:768px) {
  .c7 h1 {
    font-size: 68px;
    line-height: 74px;
  }
}

@media screen and (min-width:768px) {
  .c7 h2 {
    font-size: 56px;
    line-height: 64px;
  }
}

@media screen and (min-width:768px) {
  .c7 h3 {
    font-size: 44px;
    line-height: 50px;
  }
}

@media screen and (min-width:768px) {
  .c7 h4 {
    font-size: 32px;
    line-height: 38px;
  }
}

@media screen and (min-width:768px) {
  .c7 h5 {
    font-size: 24px;
    line-height: 28px;
  }
}

@media screen and (min-width:768px) {
  .c7 h6 {
    font-size: 18px;
    line-height: 22px;
  }
}

@media screen and (min-width:768px) {
  .c7 ul,
  .c7 ol {
    margin-bottom: l;
  }
}

@media screen and (min-width:768px) {
  .c8 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    margin-top: l;
  }
}

@media screen and (min-width:768px) {
  .c8 > * {
    margin-bottom: 0;
  }
}

@media screen and (min-width:768px) {
  .c8 > *:first-child,
  .c8 > *:last-child {
    margin-bottom: 0;
  }
}

<div>
  <div
    class="c0"
    data-gtm-category="Others"
    data-gtm-component-type="FullWidthCard"
    id="anchor-id"
  >
    <div
      class="c1"
    >
      <img
        alt="mock alt"
        class="c2"
        draggable="true"
        loading="lazy"
        src="mock.png"
      />
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
          mb="xs,m"
        >
          

          <h5
            class="c6"
          >
            Title
          </h5>
        </div>
        <div
          class="c7"
          mb="0"
          textalign="center"
        >
          

          <p>
            BodyText
          </p>
        </div>
      </div>
      <div
        class="c8"
      />
    </div>
  </div>
</div>
`;
